Environment_Engine/Query/BottomLeft.cs

```diff
@@ -63,9 +63,9 @@ public static Point BottomLeft(this Polyline polyline, List<Panel> panelsAsSpace
 
             Point centre = polyline.Centroid();
             if (centre == null)
-                centre = polyline.Centre();
+                centre = polyline.ControlPoints.CullDuplicates().Average();
 
-            if(normal == null)
+            if (normal == null)
```

Comment:
While we're here... could we move this to happen after line 62? Seems redundant to get the vector and then do a computation before returning `null` if the vector was `null`.